src/librustc/mir/mod.rs

```diff
@@ -2769,6 +2769,13 @@ pub struct BorrowCheckResult<'tcx> {
     pub used_mut_upvars: SmallVec<[Field; 8]>,
 }
 
+/// The result of the `mir_const_qualif` query.
+#[derive(Clone, Copy, Debug, Default, RustcEncodable, RustcDecodable, HashStable)]
+pub struct QualifSet {
+    pub has_mut_interior: bool,
+    pub needs_drop: bool,
```

Comment:
I wonder if these two fields should have similar doc comments to the `HasMutInterior` and `NeedsDrop` types in `rustc_mir::transform::check_consts`.

Reply:
Would a link to the `Qualif` implementers suffice? I don't wanna copy-paste lest they get out of sync.

Reply:
Oh I wasn't thinking copy-pasting, maybe a few words and a link (or even just the `rustc_mir::...::HasMutInterior` etc. path).

Reply:
I added a note on `ConstQualifs` that points the user to `librustc_mir/transform/check_consts/qualifs.rs`. Lemme know if you want something more here.